enforcer-parent/enforcer/src/main/java/org/wso2/choreo/connect/enforcer/grpc/ExtAuthService.java

```diff
@@ -140,6 +143,17 @@ private CheckResponse buildResponse(CheckRequest request, ResponseObject respons
                     .build();
         } else {
             OkHttpResponse.Builder okResponseBuilder = OkHttpResponse.newBuilder();
+
+            if (responseObject.getQueryParamsToRemove().size() > 0) {
+                String constructedPath = constructQueryParamString(responseObject.getRequestPath(),
+                        responseObject.getQueryParamMap(), responseObject.getQueryParamsToRemove());
+                HeaderValueOption headerValueOption = HeaderValueOption.newBuilder()
+                        .setHeader(HeaderValue.newBuilder().setKey(APIConstants.PATH_HEADER).setValue(constructedPath)
+                                .build())
+                        .build();
+                okResponseBuilder.addHeaders(headerValueOption);
```

Comment:
Better to add a comment here  explaining what we are doing in this piece of code and what we are trying to avoid by this.